config/v1alpha1/types_backup.go

```diff
@@ -0,0 +1,131 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//
+// Backup provides configuration for performing backups of the openshift cluster.
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type Backup struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// metadata is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec BackupSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +kubebuilder:validation:Optional
+	// +optional
+	Status BackupStatus `json:"status"`
+}
+
+type BackupSpec struct {
+	// +kubebuilder:validation:Required
+	EtcdBackupSpec EtcdBackupSpec `json:"etcdBackupSpec,omitempty"`
+}
+
+type BackupStatus struct {
+}
+
+// EtcdBackupSpec provides configuration for automated etcd backups to the cluster-etcd-operator
+type EtcdBackupSpec struct {
+
+	// Schedule defines the recurring backup schedule in Cron format
+	// every 2 hours: 0 */2 * * *
+	// every day at 3am: 0 3 * * *
+	// Setting to an empty string "" means disabling scheduled backups
+	// Default: ""
+	// TODO: Define CEL validation for the cron format
+	// See if the upstream CronJob has CEL validation already defined somewhere.
+	// TODO: Validation to disallow unrealistic schedules (e.g */2 * * * * every 2 mins)
+	// +kubebuilder:validation:Optional
+	// +optional
+	Schedule string `json:"schedule"`
+
+	// The time zone name for the given schedule, see https://en.wikipedia.org/wiki/List_of_tz_database_time_zones.
+	// If not specified, this will default to the time zone of the kube-controller-manager process.
+	// See https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#time-zones
+	// +kubebuilder:validation:Optional
+	// +optional
+	// TODO: CEL validation for timezone format
+	TimeZone string `json:"timeZone,omitempty"`
+
+	// RetentionPolicy defines the retention policy for retaining and deleting existing backups.
+	// +kubebuilder:validation:Optional
+	// +optional
+	RetentionPolicy RetentionPolicy `json: "retentionPolicy"`
+
+	// PVCName specifies the name of the PersistentVolumeClaim which binds a PersistentVolume where the
+	// etcd backup files would be saved
+	// +kubebuilder:validation:Required
+	// +required
+	// TODO: Cannot be empty string
+	PVCName string `json: "pvcName"`
```

Comment:
the PVC requires a namespace, we would create it in openshift-etcd-operator? Probably worth mentioning this in the comment, too.

Reply:
Good point I forgot about PVC being namespaced. I guess the admin would need to create it in the etcd-operator namespace so that the backup executing pod can use it. I'll add a comment.